### PR TITLE
DSD-1830: updating zindex in the FeedbackBox component

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-1830](https://newyorkpubliclibrary.atlassian.net/browse/DSD-1830)

## This PR does the following:

This issue was found in a Research Catalog page. In mobile, the `MultiSelect` components displayed above the `FeedbackBox` button and drawer components.

- This updates the z-index of the button but also Chakra's `Drawer` component. Their component, unfortunately, does not accept a class name or style props so the z-index rule is added as a global rule. This still has the same effect.
- This is difficult to test in Storybook because we don't have examples where we can easily add a `FeedbackBox` button on `MultiSelects` that makes sense. So perhaps a Turbine example is better.

## How has this been tested?

- Locally and in Research Catalog.

Issues fixed:
Button
Screenshot 2024-09-10 at 11 10 45 AM

Drawer
Screenshot 2024-09-13 at 1 57 06 PM

## Accessibility concerns or updates

-

### Accessibility Checklist

- [x] Checked Storybook's "Accessibility" tab for color contrast and other issues.
- [x] The feature works with keyboard inputs including tabbing back and forward and pressing space, enter, arrow, and esc keys.
- [x] For hidden text or when `aria-live` is used, a screenreader was used to verify the text is read.
- [x] For features that involve UI updates and focusing on DOM `ref`s, focus management was reviewed.
- [x] The feature works when the page is zoomed in to 200% and 400%.

## Open Questions

-

### Checklist:

- [x] I have updated the Storybook documentation and changelog accordingly.
- [x] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

[DSD-1830]: https://newyorkpubliclibrary.atlassian.net/browse/DSD-1830?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by 7emansell — APPROVED** [review]

File: src/theme/foundations/global.ts [hunk 1]
```diff
@@ -48,6 +48,9 @@ const global = {
   "*, *::before, &::after": {
     boxSizing: "inherit",
   },
+  ".chakra-modal__content-container": {
+    zIndex: "999999 !important",
+  },
```
Comment (bigfishdesign13) [inline src/theme/foundations/global.ts:53]: The button within the `MultiSelect` has a z-index of 10000, but it is showing above the `FeedbackBox` background.  It seems that the z-index of 1400 that is applied to the `FeedbackBox` popup element (the `role="dialog"` element)  is what is causing the problem.  And that value is coming from the `--nypl-zIndices-modal` CSS var.

image

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
The layering issue is not totally fixed.  Please see my comment.

**Comment by EdwinGuzman** [comment]
Fixed in latest update: 
Screenshot 2024-09-18 at 4 54 39 PM

**Review by bigfishdesign13 — APPROVED** [review]
Very nice.

**Pull request merged**